xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-rest/src/main/java/org/xwiki/livedata/internal/rest/DefaultLiveDataEntriesResource.java

```diff
@@ -66,25 +71,22 @@
     private static final String FILTERS_PREFIX = "filters.";
 
     @Override
-    public Entries getEntries(String sourceId, String namespace, List<String> properties, List<String> matchAll,
+    public Entries getEntries(String sourceId, String namespaceStr, List<String> properties, List<String> matchAll,
         List<String> sort, List<Boolean> descending, long offset, int limit) throws Exception
     {
-        // Workaround for https://github.com/restlet/restlet-framework-java/issues/922 (JaxRs multivalue query-params
-        // gives list with null element)
-        List<String> actualProperties = properties.stream().filter(Objects::nonNull).collect(Collectors.toList());
-        List<String> actualMatchAll = matchAll.stream().filter(Objects::nonNull).collect(Collectors.toList());
-        List<String> actualSort = sort.stream().filter(Objects::nonNull).collect(Collectors.toList());
-        List<Boolean> actualDescending = descending.stream().filter(Objects::nonNull).collect(Collectors.toList());
-        LiveDataConfiguration config =
-            getConfig(sourceId, actualProperties, actualMatchAll, actualSort, actualDescending, offset, limit);
-        Optional<LiveDataSource> source = this.liveDataSourceManager.get(config.getQuery().getSource(), namespace);
-        if (source.isPresent()) {
-            String idProperty = config.getMeta().getEntryDescriptor().getIdProperty();
-            LiveDataEntryStore entryStore = source.get().getEntries();
-            return createEntries(entryStore.get(config.getQuery()), idProperty, config.getQuery().getSource(),
-                namespace).withOffset(offset).withLimit(limit);
-        } else {
-            throw new WebApplicationException(Response.Status.NOT_FOUND);
+        XWikiContext xWikiContext = this.xcontextProvider.get();
+        String savedWikiId = xWikiContext.getWikiId();
+        try {
+            // Switch the wikiId of the context if a namespace is provided and is of type wiki (for instance "wiki:s1").
+            Namespace namespace = NamespaceUtils.toNamespace(namespaceStr);
+            if (namespace != null && Objects.equals(namespace.getType(), WikiNamespace.TYPE)) {
+                xWikiContext.setWikiId(namespace.getValue());
+            }
+
+            LiveDataConfiguration config = initConfig(sourceId, properties, matchAll, sort, descending, offset, limit);
+            return getEntries(namespaceStr, offset, limit, config);
+        } finally {
+            xWikiContext.setWikiId(savedWikiId);
```

Comment:
This rollback is not needed in practice because each request calls a single REST resource so there's nothing else executed after, that would use the current wiki.

Reply:
I understand this finally is not useful now but  couldn't this kind of call be wrapped or followed by some generic code using the context in the future?